packages/forklift-console-plugin/src/modules/Plans/views/details/components/PlanPageHeadings.tsx

```diff
@@ -44,10 +58,25 @@ export const PlanPageHeadings: React.FC<{ name: string; namespace: string }> = (
   const buttonStartIcon = canReStart ? <ReStartIcon /> : <StartIcon />;
 
   const criticalCondition =
-    loaded &&
-    !error &&
+    planLoaded &&
+    !planError &&
     plan?.status?.conditions?.find((condition) => condition?.category === 'Critical');
 
+  /**
+   * Check if the preserve static IPs is enabled with pod networking mapping.
+   */
+  const preserveIpsWithPodMapCondition = (): boolean => {
+    if (!netMapsLoaded || netMapsError) return false;
+
+    const isPreserveStaticIPs = plan?.spec?.preserveStaticIPs;
+    const planNetMaps = netMaps
+      ? netMaps.find((net) => net?.metadata?.name === plan?.spec?.map?.network?.name)
+      : null;
+    const isMapToPod = planNetMaps?.spec?.map.find((map) => map.destination.type === 'pod') != null;
```

Comment:
i dont see a point in != null (btw it should be !==) 

if find canont find an item it will be undefined and the && condition below will fail

Reply:
`isMapToPod` should be of a boolean type to avoid typying error, that's why I compare to null.
Since the value can also be `undefined` then I used ` != null`
I can do the manipulation to convert `isMapToPod` in the return value but I think it will be less clear. WDYT?

Reply:
when changing the code by omitting the comparison to null:
```
const isMapToPod = planNetMaps?.spec?.map.find((map) => map.destination.type === 'pod');

return isPreserveStaticIPs && isMapToPod;
```

I get the following TS error:

`TS2322: Type 'V1beta1NetworkMapSpecMap' is not assignable to type 'boolean'.`

So a comparison/casting should be done. therefore I changed  to 
`!== undefined`